### PR TITLE
Support `Center(::Function)`, `Scale(::Function)`, `ZScore(;center::Function, scale::Function)`

### LINKED ISSUE #19: Allow `Center(::Function)`, `Scale(::Function)`, `ZScore(::Function, ::Function)` in schema hints/contrasts
It looks like it will take a little bit more book keeping, but should be doable and would be a big convenience boost.

### PR DESCRIPTION
**Pull request by palday**
Closes #19 

- [x] `Center(::Function)`
- [x] `Scale(::Function)`
- [x] `ZScore(::Function)`
- [x] let JuliaFormatter have its YASG way

### CODE REVIEW

File: test/centering.jl [hunk 1]
```diff
@@ -14,6 +13,13 @@
         @test yc isa CenteredTerm
         @test yc.center == mean(data.y)
         @test modelcols(yc, data) == data.y .- mean(data.y) == data.y .- yc.center
+
+        @testset "alternative center function" begin
+            xc = concrete_term(term(:x), data, Center(median))
+            @test xc isa CenteredTerm
+            @test xc.center == median(data.x)
+            @test modelcols(xc, data) == data.x .- median(data.x) == data.x .- xc.center
```
Comment (kleinschmidt) [inline test/centering.jl:21]: in this case median is going to give you the same as mean right?  maybe use something like `first` to really get wacky?

**Review by kleinschmidt — COMMENTED** [review]
looks good, could maybe beef up the tests a bit

File: test/centering.jl (hunk 1 above)
Comment (palday) [reply test/centering.jl:21]: That's a good idea.

File: test/centering.jl [hunk 2]
```diff
@@ -15,10 +15,15 @@
...
+            xc = concrete_term(term(:x), data, Center(f))
             @test xc isa CenteredTerm
-            @test xc.center == median(data.x)
-            @test modelcols(xc, data) == data.x .- median(data.x) == data.x .- xc.center
+            @test xc.center == f(data.x)
+            @test modelcols(xc, data) == data.x .- f(data.x) == data.x .- xc.center
+            # why test this? well this makes sure that our tests
+            # wouldn't pass if we were using the default center function
+            # in other words, this tests we're actually hitting a different codepath
+            @test !isapprox(mean(data.x), f(data.x))
```
Comment (kleinschmidt) [inline test/centering.jl:26]: :kissing: :ok_hand:

**Review by kleinschmidt — APPROVED** [review]
looks great, thanks!!

**Pull request merged**